python/pyspark/sql/tests/connect/test_connect_basic.py

```diff
@@ -389,28 +389,30 @@ def test_with_local_list(self):
             self.connect.createDataFrame(data, "col1 int, col2 int, col3 int").show()
 
     def test_with_local_rows(self):
-        # SPARK-41789: Test creating a dataframe with list of Rows
-        data = [
-            Row(course="dotNET", year=2012, earnings=10000),
-            Row(course="Java", year=2012, earnings=20000),
-            Row(course="dotNET", year=2012, earnings=5000),
-            Row(course="dotNET", year=2013, earnings=48000),
-            Row(course="Java", year=2013, earnings=30000),
-            Row(course="Scala", year=2022, earnings=None),
+        # SPARK-41789, SPARK-41810: Test creating a dataframe with list of rows and dictionaries
+        rows = [
+            Row(course="dotNET", earnings=10000, year=2012),
+            Row(course="Java", earnings=20000, year=2012),
```

Comment:
The test data had to be fixed because PySpark sorts the names from dictionaries when inferring the schema.